core/src/main/java/discord4j/core/object/audit/AuditLogEntry.java

```diff
@@ -103,8 +103,12 @@ public Optional<Snowflake> getResponsibleUserId() {
      * Gets the user who made the changes, if present.
      *
      * @return The user who made the changes, if present.
+     * @throws NoSuchElementException if the {@link AuditLogEntry#getParent()} not contains the {@link #getResponsibleUserId()} or is null.
      */
     public Optional<User> getResponsibleUser() {
+        if (auditLogPart == null) {
+            throw new NoSuchElementException("The AuditLogPart is not available for this Audit log.");
+        }
```

Comment:
What if D4J could return that cached user if `auditLogPart == null`. The entry has a reference to GatewayDiscordClient and could fetch it

What do you think?

Reply:
the cached is not a Flux? i dont know how parse this into a Optional.

Reply:
Yeah, that's unfortunate. Maybe in a future branch

If this only happens with gateway events, how about we mention this in the exception and javadoc?

Something like: "Audit log entries from gateway events do not include User details, call #getResponsibleUserId instead"

Reply:
i mean the throw in javadoc mention this (the @throw) but if is necesary can mention this in better by the NoSuchElementException? i add a minor doc change in header for mention this.